jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -1092,7 +1092,8 @@ public void earlyEOF()
                     stream._chunk = Content.Chunk.from(bad);
                 }
 
-                ThreadPool.executeImmediately(getServer().getThreadPool(), _httpChannel.onFailure(bad));
+//              Case 1
+//                ThreadPool.executeImmediately(getServer().getThreadPool(), _httpChannel.onFailure(bad));
```

Comment:
@gregw said: _I think we should notify EOF as a failure._

Reply:
As discussed with @sbordet, `earlyEOF()` can only be invoked when the user-code does a read, hence setting the `stream._chunk` to an error chunk is enough to notify the user-code that something bad happened (the error chunk would be returned by Core, the held exception thrown by EE*).

Spawning a thread to call the `onFailure` path would notify the failure listener (in Core; which would call `AsyncListener.onError()` on EE*) in parallel to the blocking read, hence delivering the same error twice by two concurrent threads.